### PR TITLE
Bump ubuntu to 23.04

### PR DESCRIPTION
**Pull request by ben-harvey**
I had to rework the `Dockerfile` a bit to get this to work.  Can't seem to find any documentation for it, but it looks like there were convenience wrappers around `groupadd` (`addgroup`)  and `useradd` (`adduser`) that were removed between versions.  So this change just calls them directly.  Also, I had to use a new group ID since `1000` is taken now.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,9 +13,9 @@ RUN \
     && mkdir runner \
     && tar xzf "actions-runner-linux-x64-${ACTIONS_VERSION}.tar.gz" --directory ./runner
 
-FROM ubuntu:22.10
+FROM ubuntu:23.04
 
-RUN addgroup --gid 1000 "runner" && adduser --uid 1000 --ingroup "runner" --shell /bin/bash "runner" \
+RUN groupadd --gid 1234 "runner" && useradd -g "runner" --shell /bin/bash "runner" \
```
Comment (crbudzeak) [inline Dockerfile:18]: For the group ID, did 1000 not work on ubuntu 23? I think they were following the standard for this:

> By default, the new group gets created with a group id higher than the GID_MIN value defined in /etc/login.defs. On most Linux system, this value is 1000.
> In other words, you get one of the first available GID after 1000.

1000 might not work for a container you are logged into vs being built by Docker.

**Review by crbudzeak — COMMENTED** [review]
Thanks for fixing this. I have a couple of questions.

File: .github/workflows/docker-build.yml [hunk 2]
```diff
@@ -85,7 +85,7 @@ jobs:
           docker run --rm \
           -e REPO_OWNER=Enterprise-CMCS \
           -e REPO_NAME=github-actions-runner-aws \
-          -e PERSONAL_ACCESS_TOKEN=${{ secrets.BHARVEY_GITHUB_TOKEN}} \
+          -e PERSONAL_ACCESS_TOKEN=${{ secrets.BHARVEY_GITHUB_TOKEN }} \
```
[the remark this answers is not in the document]
Comment (ben-harvey) [reply .github/workflows/docker-build.yml:88]: Yep

File: Dockerfile (hunk 1 above)
Comment (ben-harvey) [reply Dockerfile]: Per our Slack conversation, we'll just skip specifying the group ID since we don't need to reference it anywhere else

**Review by crbudzeak — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**